src/js/components/AppListComponent.jsx

```diff
@@ -314,22 +366,21 @@ var AppListComponent = React.createClass({
       appsHealthCount: getInitialFilterCounts(HealthStatus)
     };
 
-    // Global search view - only display filtered apps
+    // Global search view - only display filtered apps & groups
     if (this.hasFilters()) {
       appListViewType = AppListViewTypes.APP_LIST;
 
-      nodesSequence = this.filterNodes(lazy(state.apps), filterCounts)
+      nodesSequence = lazy(this.filterNodes(state.apps, filterCounts))
```

Comment:
Hi hi! :)

May I ask which intention is behind the move of lazy.js outside of `filterNodes`?
Inside of `filterNodes` is a very long filter sequence chain that gets lazy evaluated.
This isn't the case anymore with this change and probably a performance regression.